### ISSUE TITLE
Obsolete terms showing in annotation extension term selector

### ISSUE THREAD

**Issue by jseager7** [pombase/canto#2570]
We recently noticed that some terms that are marked obsolete in PHIPO_EXT (an ontology we use for annotation extension terms) are showing up in the term selector:

![image](https://user-images.githubusercontent.com/37659591/167815313-3ea3ab6c-198c-47dd-be0d-f132281de2e0.png)

I'm guessing this is an issue with the PHIPO_EXT ontology, because I've compared obsolete terms in FYPO and PHIPO to PHIPO_EXT and noticed that the obsolete terms in these other ontologies have had their `is_a` relation removed:

```
[Term]
id: FYPO:0000048
name: obsolete invasive growth abolished
alt_id: FYPO:0000465
def: "OBSOLETE. A cellular process phenotype in which invasive growth does not occur." [PomBase:mah]
comment: This term was made obsolete because it is not required to describe fission yeast.
synonym: "absent invasive growth" EXACT [PomBase:mah]
synonym: "invasive growth abolished during vegetative growth" EXACT [PomBase:mah]
synonym: "invasive growth: absent" RELATED [SGD:phenotype_annotation]
is_obsolete: true
consider: FYPO:0000551
created_by: midori
creation_date: 2011-03-29T05:53:56Z
```

Whereas the `is_a` relation is still there in PHIPO_EXT:

```
[Term]
id: PHIPO_EXT:0000015
name: obsolete compatible interaction, recognizable pathogen effector present, functional host resistance gene present
def: "A pathogen host interaction phenotype where the gene-for-gene relationship between the pathogen and plant host results in disease susceptibility. The ability of a pathogen effector to be recognized by a specific host resistance gene is present and the ability of a host to be resistant to a pathogen, via the recognition of a pathogen effector by a specific host resistance gene, is present." []
subset: qc_extension_only
is_a: PHIPO_EXT:0000001 ! gene-for-gene interaction phenotype
is_obsolete: true
created_by: alaynecuzick
creation_date: 2020-11-11T12:59:19Z
```

- - -

@kimrutherford I'm guessing that keeping the `is_a` relation keeps the term out of the `canto_root_subset` and doesn't allow it to be filtered out by the `ontology_namespace_config` (our config is shown below). Is this correct?

```yaml
ontology_namespace_config:
  subsets_to_ignore:
    primary_autocomplete:
      - "is_a(Grouping_terms)"
      - "is_a(qc_extension_only)"
    primary_select:
      - "is_a(Grouping_terms)"
      - "is_a(qc_do_not_annotate)"
      - "is_a(qc_extension_only)"
    extension:
      - "is_a(canto_root_subset)"
      - "is_a(qc_do_not_annotate)"
  do_not_annotate_subsets:
    - "is_a(canto_root_subset)"
    - "is_a(qc_do_not_annotate)"
    - "is_a(gocheck_do_not_annotate)"
    - "is_a(gocheck_do_not_manually_annotate)"
```

**Comment by kimrutherford** [comment]
> I'm guessing that keeping the is_a relation keeps the term out of the canto_root_subset and doesn't allow it to be filtered out by the ontology_namespace_config (our config is shown below). Is this correct?

Hi James.

Yes, I think you have it right.  To be honest I didn't realise that's how the obsolete terms were being hidden in Canto - it's been a long time since I looked at that code.  All the obsolete terms in FYPO and GO have no is_a so I hadn't noticed this.

The obsolete flag for each term is stored in Canto's database but it isn't used to deciding which terms should be displayed.  We do use that flag to highlight obsolete terms in red in the tables:
![obsolete-1](https://user-images.githubusercontent.com/90474/167992959-432ec217-12ce-4767-a6b9-b6a8e00a8dac.png)

I think the code should be obeying the obsolete flag for term selection, but it will be a while before I have time to look into that.  So the quick fix for PHIPO_EXT is to remove the is_a relations for the obsolete terms.

I made a separate issue about fixing the code: #2571
